light-node/src/commands/start.rs

```diff
@@ -4,94 +4,106 @@
+                    peer_list = peer_list.primary(instance.light_client.peer, instance);
+                } else {
+                    peer_list = peer_list.witness(instance.light_client.peer, instance);
+                }
+            }
+            let peer_list = peer_list.build();
 
-            let vals_hash = Hash::from_hex_upper(
-                hash::Algorithm::Sha256,
-                &config.subjective_init.validators_hash,
-            )
-            .unwrap();
+            let mut supervisor = Supervisor::new(
+                peer_list,
+                ProdForkDetector::default(),
+                ProdEvidenceReporter::new(peer_map.clone()),
+            );
 
-            println!("Requesting from {}.", config.rpc_address);
+            let handle = supervisor.handle();
+            let rpc_handler = supervisor.handle();
+            let server = Server::new(rpc_handler);
 
-            subjective_init(config.subjective_init.height, vals_hash, &mut store, &req)
-                .await
-                .unwrap();
+            std::thread::spawn(|| supervisor.run());
+            let laddr = app_config().rpc_config.listen_addr;
+            // TODO: figure out howto handle the potenial error on run
+            std::thread::spawn(move || rpc::run(server, &laddr.to_string()));
 
             loop {
-                let latest_sh = (&req).signed_header(0).await.unwrap();
-                let latest_peer_height = latest_sh.header().height();
-
-                let latest_trusted = store.get(0).unwrap();
-                let latest_trusted_height = latest_trusted.last_header().header().height();
-
-                // only bisect to higher heights
-                if latest_peer_height <= latest_trusted_height {
-                    std::thread::sleep(Duration::new(1, 0));
-                    continue;
+                match handle.verify_to_highest() {
+                    Ok(light_block) => {
+                        println!("[info] synced to block {}", light_block.height());
+                    }
+                    Err(err) => {
+                        println!("[error] sync failed: {}", err);
+                    }
                 }
 
-                println!(
-                    "attempting bisection from height {:?} to height {:?}",
-                    latest_trusted_height, latest_peer_height,
-                );
-
-                let now = SystemTime::now();
-                lite::verify_bisection(
-                    latest_trusted.to_owned(),
-                    latest_peer_height,
-                    TrustThresholdFraction::default(), // TODO
-                    config.trusting_period,
-                    now,
-                    &req,
-                )
-                .await
-                .unwrap();
-
-                println!("Succeeded bisecting!");
-
-                // notifications ?
-
-                // sleep for a few secs ?
+                std::thread::sleep(Duration::from_millis(800));
```

Comment:
Why do we have this random sleep here?

Reply:
Hmm, that needs more thoughts: when catching up to the current height, sleeping doesn't make sense here. But as soon as the light client caught up, it does not really make sense to request the latest light block several times before a new block is created 🤔 

On the command line this will look like:
```
[info] synced to block 830
[info] synced to block 831
[info] synced to block 831
[info] synced to block 831
[info] synced to block 831
[info] synced to block 831
[info] synced to block 831
[info] synced to block 831
[info] synced to block 831
```

Reply:
What's the behaviour we want? That it syncs on an interval? Or that it loops as quickly until a certain condition is met?

Reply:
My understanding is: if lagging behind, catch up without delay as quickly as possible. If caught up, try sync every blocktime/2 or sth like that?

Reply:
Would be nice to surface that behaviour explicitly. And in the synced state use a timer, for example from tokio, are we using pure threads in the light-node?

Reply:
Or I think we can use a [timer](https://docs.rs/crossbeam-channel/0.4.2/crossbeam_channel/fn.tick.html) if it's lighter.

Reply:
Thanks, good minimal solution right there.

Reply:
`crossbeam_channel::tick` looks like a good fit but I'm wondering if we should stick to `std::thread::sleep` as it doesn't require yet another dependency. LMK what you think.

Reply:
The only solution is fine for now, we can look into `crossbeam` once we have more components to coordinate.